### ISSUE TITLE
Not all events are sent — How to debug?

### ISSUE THREAD

**Issue by tgowing** [managed-components/facebook-pixel#12]
First of all, thank you so much for this integration!

**TL;DR**: Not all the Zaraz events triggering Facebook Pixel are arriving on the Facebook end. They get lost somehere in between Zaraz and Facebook. 

---

We have the following case: we send client side requests (using the Facebook Pixel) and server side events (using Zaraz + this component) for analytics. They are deduplicated by the the event_id.

However, in the Facebook dashboard we see more client side events than server side, which is unexpected. Assuming that browsers might block the client side requests, it should be the oposite. The difference for example is about 650 events are sent recieved via the Facebook pixel and only about 450 recieved by Facebook via the Zaraz server side.

Going deeper in the debugging, the exact same function from the code on the website sends the conversion event to both Facebook directly (pixel) and to Zaraz. - from the 'monitoring' section in the zaraz dashboard we can see that the correct number of events are recieved by Zaraz ie 650 (the same as are sent to Facebook via the pixel). But when we check in the Facebook Events interface we can see that there are only 450 events are sent from the server (ie Zaraz). So somewhere between Zaraz recieving the events and Facebook recieving them some (about a third) of them are lost.

Since there are no logs in Zaraz, it is hard to debug it between Zaraz and Facebook. Any ideas how we could handle this?

Thanks!

**Comment by Refaerds** [comment]
hi @tgowing thanks for reporting this! We're already investigating. Would you mind sharing your zone name (website)?

**Comment by tgowing** [comment]
@Refaerds Sure, but sorry for my ignorance but what exactly is a zone name and where do I find it? A zone id?

**Comment by Refaerds** [comment]
@tgowing your website url should be enough. Thanks in advance!

**Comment by tgowing** [comment]
humancondition.com

**Comment by tgowing** [comment]
@Refaerds It looks like you have fixed the issue. Thanks very much!
![Screenshot 2023-09-30 at 7 43 13 am](https://github.com/managed-components/facebook-pixel/assets/11142114/9967da48-7b94-4a1e-8e2e-d222eebe5898)

**Comment by Refaerds** [comment]
Thanks a lot for confirming @tgowing ! We'll release the bug fix for all users soon
